lib/fog/openstack/requests/network/create_subnet.rb

```diff
@@ -14,7 +14,7 @@ def create_subnet(network_id, cidr, ip_version, options = {})
           vanilla_options = [:name, :gateway_ip, :allocation_pools,
                              :dns_nameservers, :host_routes, :enable_dhcp,
                              :tenant_id]
-          vanilla_options.reject{ |o| options[o].nil? unless o == :gateway_ip }.each do |key|
+          vanilla_options.select{ |o| options.key?(o) }.each do |key|
```

Comment:
hm, might be good to extract this behavior to some helper, cause I think I saw this filtering on other places too. 

Should be always .key?, users should be free to send any value they need.

But that is not scope of this PR, maybe if you will be bored in the future, or somebody else can tackle it. :-)